### PR TITLE
feat(addon-editor): add `TUI_EDITOR_MAX_IMAGE_WIDTH` token

### LINKED ISSUE #2363: 🚀 - Возможность выбора способа добавления картинки в редактор
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

_No response_

### Description

Хотелось бы иметь возможность кастомизировать добавление картинки в редактор не только в base64 формате, но и переопределить своё собственное поведение

Пример кейса:
- пользователь нажимает ctrl + v / выбирает картинку через тулбарную кнопку
- вызывается переопределённая функция, в которой 
- отправляем картинку в облако
- получаем url и вставляем в редактор img с src картинки 

Также, токен TUI_EDITOR_CONTENT_PROCESSOR не реагирует на изменения эдитора, возможно, так не должно быть

ИМХО, нам было бы удобнее использовать токен, чтобы в аутпуты компонента сыпалось всё как сейчас, а по токену можно было подписать на загрузку файлов и её настраивать.

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [x] Documentation content changes

## What is the current behavior?

Closes #2363

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/186888345-a08b830e-c11c-43c4-b974-140bccae39aa.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/addon-editor/extensions/image-editor/image-editor.component.ts [hunk 1]
```diff
@@ -14,6 +14,13 @@ export const TUI_EDITOR_MIN_IMAGE_WIDTH = new InjectionToken<number>(
     },
 );
 
+export const TUI_EDITOR_MAX_IMAGE_WIDTH = new InjectionToken<number | null>(
+    `Max size of resizable image inside editor`,
+    {
+        factory: () => null,
```
Comment (nsbarsukov) [inline projects/addon-editor/extensions/image-editor/image-editor.component.ts:20]:
```ts
export const TUI_EDITOR_MAX_IMAGE_WIDTH = new InjectionToken<number>(
    `Max size of resizable image inside editor`,
    {
        factory: () => Infinity,
    },
);
```

**Review by waterplea — APPROVED** [review]

File: projects/demo/src/modules/components/editor-new/examples/7/imgbb.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    success: boolean;
+    status: number;
+}
+
+@Injectable({
+    providedIn: `root`,
+})
+export class ImgbbService {
+    private readonly apiHost = `https://api.imgbb.com`;
+    private readonly apiKey = `3c1615980dcf693b282c4b0fb608b28a`;
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/demo/src/modules/components/editor-new/examples/7/imgbb.service.ts:25]: or should we save it as environment variable ?

[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/editor-new/examples/7/imgbb.service.ts]: fixed

File: projects/addon-editor/extensions/image-editor/image-editor.component.ts [hunk 3]
```diff
@@ -46,7 +53,8 @@ export class TuiImageEditorComponent extends TuiNodeViewNgComponent {
     }
 
     constructor(
-        @Inject(TUI_EDITOR_MIN_IMAGE_WIDTH) private readonly minWidth: number,
+        @Inject(TUI_EDITOR_MIN_IMAGE_WIDTH) readonly minWidth: number,
+        @Inject(TUI_EDITOR_MAX_IMAGE_WIDTH) readonly maxWidth: number | null,
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/extensions/image-editor/image-editor.component.ts]: fixed

**Pull request merged**